frontend/src/metabase/query_builder/components/notebook/steps/JoinStep/JoinStep.tsx

```diff
@@ -53,19 +53,7 @@ export function JoinStep({
   } = useJoin(query, stageIndex, join);
 
   const [isAddingNewCondition, setIsAddingNewCondition] = useState(false);
-
-  // Is only needed for `joinLHSDisplayName`
-  // to properly display the LHS table name until the first condition is complete
-  const [selectedLHSColumn, setSelectedLHSColumn] = useState<
-    Lib.ColumnMetadata | undefined
-  >();
-
-  const lhsDisplayName = Lib.joinLHSDisplayName(
-    query,
-    stageIndex,
-    join || table,
-    selectedLHSColumn,
-  );
+  const [lhsDisplayName, setLhsDisplayName] = useState("");
```

Comment:
The change looks hacky. The display name is definitely not a "state". I think e2e tests make sense however the change should be somewhat different. Let's discuss it

Reply:
Reverted the change: 82dacaf8c1a5750c27bd9e7432e99c752579a9b2

We discussed it on a call and this hacky change should not be necessary once the real source of the issue is fixed in the backend (MLv2). But we'd need to double-check this later after backend is fixed (tests added in this PR should fail).